src/main/java/com/techiteasy/demo/controllers/ExceptionController.java

```diff
@@ -0,0 +1,30 @@
+package com.techiteasy.demo.controllers;
+
+import com.techiteasy.demo.exceptions.RecordNotFoundException;
+import com.techiteasy.demo.exceptions.TelevisionNameTooLongException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+//Deze klasse behandelt uitzonderingen (zoals de RecordNotFoundException)
+@Controller // Stap 5: voorzie deze klasse van de juiste annotatie
+public class ExceptionController {
+
+    // ExceptionHandler voor RecordNotFoundException
+    @ExceptionHandler(RecordNotFoundException.class)
+    public ResponseEntity<String> handleRecordNotFoundException(RecordNotFoundException ex) {
+        return ResponseEntity.status(404).body(ex.getMessage());
+    }
+    // ExceptionHandler voor IndexOutOfBoundsException (bij een ongeldige index in de mock-database)
+    @ExceptionHandler(TelevisionNameTooLongException.class)
+    //public ResponseEntity<String> handleIndexOutOfBoundsException(IndexOutOfBoundsException ex) {
+    public ResponseEntity<String> handleTelevisionNameTooLongException(IndexOutOfBoundsException ex) {
+        return ResponseEntity.status(400).body(ex.getMessage());
+    }
+
+    // ExceptionHandler voor algemene RuntimeExceptions
+    @ExceptionHandler(RuntimeException.class)
```

Comment:
Ik zou geen exceptionhandler maken voor RuntimeException. Elke exception die tijdens het runnen kan voorkomen is een RuntimeException. Zo word deze ExceptionHandler dus altijd getriggered terwijl het misschien niet iets is wat je naar de gebruiker wilt communiceren. 

Je kunt wel ExceptionHandlers maken voor ingebouwde java of spring exceptions, zoals NullPointerException of IndexOutOfBoundsException. Meestal kom je er tijdens het bouwen achter dat er een bepaalde exception kan voorkomen in een bepaalde situatie. Daar kun je dan vervolgens een exceptionhandler voor bouwen die dat netjes afhandeld. 

Last but not least... Je retuned in deze exceptionhandler een 500 status, terwijl we eigen juist die 500's willen voorkomen. Als er echt iets aan de server kant mis gaat, er gaat bijvoorbeeld een service offline of de database is corrupt, dan kun je een 500 returnen, maar dan heb je ook echt een probleem. In de meeste gevallen is de error echter veroorzaakt door de gebruiker en moet de gebruiker gewoon even diens input of acties aanpassen (gewoon ff normaal doen). Dat is dan een 400-nogwat error en daarbij geef je ook een duidelijk bericht mee waarin staat wat de gebruiker anders moet doen om het probleem op te lossen.